core/src/main/java/com/linecorp/armeria/server/annotation/decorator/RequestTimeout.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.annotation.decorator;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.concurrent.TimeUnit;
+
+import com.linecorp.armeria.server.annotation.DecoratorFactory;
+
+/**
+ * Annotation for request timeout
+ */
+@DecoratorFactory(RequestTimeoutDecoratorFunction.class)
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.TYPE, ElementType.METHOD })
+public @interface RequestTimeout {
+
+    /**
+     * Value of request timeout to set
+     */
+    long value();
+
+    /**
+     * Time unit of request timeout to set
+     */
+    TimeUnit unit();
```

Comment:
- How about setting `TimeUnit.MILLISECONDS` as the default unit?
- `TimeoutMode` could be also a useful parameter. `TimeoutMode.SET_FROM_START` may be used for the default mode.

Reply:
would adding TimeoutMode make sense given that the user doesn't know the underlying implementation? Setting TimeoutMode.SET_FROM_NOW or EXTEND alters the timeout at the point where setRequestTimeout is called(in the new decorator), which is hidden from users

Reply:
Good point. `TimeoutMode` seems to make use of `@RequestTimeout` complex. Please ignore the comment about `TimeoutMode`.